### PR TITLE
[PKG-4493] 4.6.3

### PR DESCRIPTION
**Pull request by ViridianMelody**
pymongo 4.6.3

**Destination channel:** defaults

### Links

- [PKG-4493] 
- [Upstream repository](https://github.com/mongodb/mongo-python-driver)

### Explanation of changes:

- Expanded upstream tests to all platforms.

[PKG-4493]: https://anaconda.atlassian.net/browse/PKG-4493?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -26,30 +25,34 @@ requirements:
     - setuptools
   run:
     - python
+    - dnspython >=1.16.0,<3.0.0
 
 test:
   source_files:
     - .
```
Comment (cbouss) [inline recipe/meta.yaml]:
```suggestion
    - test
    - pytest.ini
```
We can do without bringing in the whole source code.

File: recipe/meta.yaml [hunk 2]
```diff
@@ -26,30 +25,34 @@ requirements:
     - setuptools
   run:
     - python
+    - dnspython >=1.16.0,<3.0.0
```
Comment (cbouss) [inline recipe/meta.yaml:28]:
```suggestion
    - dnspython >=1.16.0,<3.0.0
  run_constrained:
    - pymongo-auth-aws <2.0.0
    - pymongocrypt >=1.6.0,<2.0.0
    - winkerberos >=0.5.0 # [win]
    - pyopenssl >=17.2.0
    - requests <3.0.0
    - cryptography >=2.5
    - service_identity >=18.1.0
```
https://github.com/mongodb/mongo-python-driver/blob/4.6.3/pyproject.toml#L46

File: recipe/meta.yaml [hunk 3]
```diff
@@ -26,30 +25,34 @@ requirements:
...
   imports:
     - bson
     - gridfs
     - pymongo
-  requires:  # [not win]
-    - mongodb  # [not win]
-    - setuptools  # [not win]
-    - pip         #[not win]
+  requires:
+    - setuptools
```
Comment (cbouss) [inline recipe/meta.yaml]:
```suggestion
```

File: recipe/meta.yaml [hunk 4]
```diff
@@ -26,30 +25,34 @@ requirements:
...
+    - pip
+    - pytest
```
Comment (cbouss) [inline recipe/meta.yaml]:
```suggestion
    - pytest >=7
```

**Review by danpetry — APPROVED** [review]

**Review by cbouss — APPROVED** [review]

File: recipe/meta.yaml (hunk 1 above)
Comment (ViridianMelody) [reply recipe/meta.yaml]: For some reason, I didn't realize you could pull in individual files.

**Pull request merged**